src/Currency/Cache/Service.php

```diff
@@ -0,0 +1,95 @@
+<?php
+
+namespace PodPoint\I18n\Currency\Cache;
+
+use Carbon\Carbon;
+use Illuminate\Contracts\Cache\Repository as Cache;
+use Illuminate\Support\Collection;
+use PodPoint\I18n\Currency\Service as CurrencyService;
+use PodPoint\I18n\Currency\ExchangeRate;
+use PodPoint\I18n\Currency\Service as ServiceInterface;
+
+class Service implements CurrencyService
+{
+    /**
+     * @var Cache
+     */
+    private $cache;
+
+    /**
+     * @var ServiceInterface
+     */
+    private $service;
+
+    /**
+     * @param Cache $cache
+     * @param ServiceInterface $service
+     */
+    public function __construct(Cache $cache, ServiceInterface $service)
+    {
+        $this->cache = $cache;
+        $this->service = $service;
+    }
+
+    /**
+     * Fetches the exchange rates from another service if they don't already exist in cache.
+     *
+     * @param string $base
+     * @param string[] $currencies
+     * @param Carbon|null $timestamp
+     *
+     * @return Collection|ExchangeRate[]
+     */
+    public function getExchangeRates(string $base = 'GBP', array $currencies = [], Carbon $timestamp = null): Collection
+    {
+        if (!$timestamp) {
+            $timestamp = Carbon::now();
```

Comment:
Can't you set this default value in the function argument? I can't remember PHP 😂

Reply:
> Expression is not allowed as a default value

🖕